pkg/metrics/interfaces.go

```diff
@@ -156,6 +157,24 @@ func (g *gauge) Opts() metricpkg.Opts { return metricpkg.Opts{} }
 
 // GaugeVec
 
+type gaugeDeletableVec struct {
```

Comment:
Is it possible to define all this on `metricpkg.Vec` so it doesn't need to be implemented for every metric type individually?

Reply:
Im not sure that makes sense when all underlying metrics with labels can be deleted in prometheus/client-golang. Whether or not we want to distinguish is a different question, but in practice, the underlying API supports it, so I think we should just surface/forward that.

Reply:
I had looked into this but noticed we already discriminate between Vec and DeletableVec. Tho prom/vec does support delete labels operations, we found it necessary to further qualify in our implementation, so figured best to follow that lead...

Reply:
Yeah, I'm not familiar with the current cilium metrics subsystem so I'll defer to you and others on this.

Reply:
I think I'm leaning towards not having a specific type. While there is some precedent for having different implementations, my guess was that there was an explicit need. In this case, I don't see an explicit need. I don't think developers will all of sudden start introducing metric deletion logic for arbitrary metrics so it feels like overengineering to protect against something that we don't know is a problem. I'd bet that most developers didn't know you could even do this to begin with (like me) :smile:.

Reply:
I understand the concerns, but think tho a 're-alignement' is in order it would be a bit more intrusive to implement given the current distinctions between Vec and DeletableVec . Hence figured this is the straighter shot for this PR's scope that would allow us to back port the fix.